templates/master/00-master/alibabacloud/files/opt-libexec-openshift-alibabacloud-routes-sh.yaml

```diff
@@ -129,7 +133,30 @@ contents:
         done
     }
 
+    remove_processed_files() {
+        # Remove processed v4vips
+        for vip in "${!v4vips[@]}"; do
+            echo "removing processed file: ${RUN_DIR}/${vip}.up"
+            rm -f "${RUN_DIR}/${vip}.up"
+        done
+
+        # Remove processed v6vips
+        for vip in "${!v6vips[@]}"; do
+            echo "removing processed file: ${RUN_DIR}/${vip}.up"
+            rm -f "${RUN_DIR}/${vip}.up"
+        done
+
+        # Remove all .down files
+        shopt -s nullglob
+        for file in "${RUN_DIR}"/*.down ; do
+            echo "removing processed file: ${file}"
+            rm -f "${file}"
+        done
+        shopt -u nullglob
+    }
```

Comment:
> If we remove the upfiles and the service runs again after because one of the vips changed state, we might think the other vips are stale when they are not. I guess we are not concerned about this in the basis that api-watcher only monitors a single URI and writes the state off all vips even if just one of them changed.

apiserver-watcher creates files for every vip on state change, this means that if there are multiple VIPs and the healthcheck fails it would create a `.down` file for each vip. It is not possible for only one VIP to change state.

This PR is trying to solve the fact that the service runs in a loop, I do not see the point of redesigning the service/script since it is meant to be used with apiserver-watcher and not as a generic, reusable service.